lib/galaxy/tools/__init__.py

```diff
@@ -804,6 +807,20 @@ def tests( self ):
             self.__tests_populated = True
         return self.__tests
 
+    def tool_provided_metadata(self, job_wrapper):
+        meta_file = os.path.join( job_wrapper.tool_working_directory, self.provided_metadata_file )
+        # LEGACY: Remove in 17.XX
+        if not os.path.exists( meta_file ):
+            # Maybe this is a legacy job, use the job working directory instead
+            meta_file = os.path.join( job_wrapper.working_directory, self.provided_metadata_file )
```

Comment:
This is left over code from last year yeah - when we separated the "job directory" from the tool working directory "job_directory/working" I left this conditional in so that Galaxy would be able to find galaxy.json files for tools launched prior to the upgrade but finished after the upgrade.